w09h03/test/pgdp/filter/DGTest.java

```diff
@@ -0,0 +1,5539 @@
+                }
+            }
+        }
+    }
+
+    @Test
+    @DisplayName("Test DG 123 VideoContainer Limit noot auf 1 Frames")
+    public void test123() throws FileNotFoundException, IllegalVideoFormatException {
+
+        int length = 1;
+
+        FrameProvider fp = new FrameProvider("noot.mp4");
+        FrameProvider fp1 = new FrameProvider("noot.mp4");
+        VideoContainer in = new VideoContainer(fp);
+        VideoContainer in1 = new VideoContainer(fp1);
+
+        Object[] array = in1.frameStream.limit(length).toArray();
+
+        // limitiere Laufzeit
+        in.limit(length);
+
+        Object[] output = in.frameStream.toArray();
+
+        Assertions.assertEquals(output.length, length);
+
+        for(int i = 0; i < length; i++){
+            for(int j = 0; j < fp.getWidth(); j++){
+                for(int k = 0; k < fp.getHeight(); k++){
+                    Assertions.assertEquals(((Frame) array[i]).getPixels().getRGB(j,k), ((Frame) output[i]).getPixels().getRGB(j,k));
+                }
+            }
+        }
+    }
+
+    @Test
+    @DisplayName("Test DG 124 VideoContainer Limit noot auf 0 Frames")
+    public void test124() throws FileNotFoundException, IllegalVideoFormatException {
+
+        int length = 0;
+
+        FrameProvider fp = new FrameProvider("noot.mp4");
+        FrameProvider fp1 = new FrameProvider("noot.mp4");
+        VideoContainer in = new VideoContainer(fp);
+        VideoContainer in1 = new VideoContainer(fp1);
+
+        // limitiere Laufzeit
+
+        try{
+            in.limit(length);
+        }
+        catch(Exception e){
+            Assertions.fail();
+        }
+
+
+        Object[] output = in.frameStream.toArray();
+
+        Assertions.assertEquals(output.length, length);
+    }
+
+    @Test
+    @DisplayName("Test DG 125 VideoContainer Limit noot auf -1 Frames")
+    public void test125() throws FileNotFoundException, IllegalVideoFormatException {
+
+        int length = -1;
+
+        FrameProvider fp = new FrameProvider("noot.mp4");
+        FrameProvider fp1 = new FrameProvider("noot.mp4");
+        VideoContainer in = new VideoContainer(fp);
+        VideoContainer in1 = new VideoContainer(fp1);
+
+        // limitiere Laufzeit
+        try{
+            in.limit(length);
+        }
+        catch(Exception e){
+            Assertions.fail();
+        }
+    }
+
+    @Test
+    @DisplayName("Test DG 126 VideoContainer Limit noot auf -10 Frames")
+    public void test126() throws FileNotFoundException, IllegalVideoFormatException {
+
+        int length = -6;
+
+        FrameProvider fp = new FrameProvider("noot.mp4");
+        FrameProvider fp1 = new FrameProvider("noot.mp4");
+        VideoContainer in = new VideoContainer(fp);
+        VideoContainer in1 = new VideoContainer(fp1);
+
+        // limitiere Laufzeit
+        try{
+            in.limit(length);
+        }
+        catch(Exception e){
+            Assertions.fail();
```

Comment:
Same here.

Reply:
We should never throw exceptions, except they are required. That was the clear instruction, thus I think this is acceptable.We should never throw exceptions, except they are required. That was the clear instruction, thus I think this is acceptable.

Reply:
> We should never throw exceptions, except they are required. That was the clear instruction, thus I think this is acceptable.We should never throw exceptions, except they are required. That was the clear instruction, thus I think this is acceptable.

https://zulip.in.tum.de/#narrow/stream/1505-PGdP-W09H03/topic/Exception.20f.C3.BCr.20invalid.20Input/near/879055

Reply:
The tests with negative length remains valid as integer parameters may have any value.